week_6/src/main/scala/org/epfl/ParallelOrNotParallel.scala

```diff
@@ -0,0 +1,64 @@
+package org.epfl.parallel
+
+import scala.concurrent.{ Await, Future}
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.duration.Duration.Inf
+
+/**
+ *  This code compares different ways to kick off two calculations
+ *  in parallel. It also shows a counter example where a for
+ *  comprehension is used in a naive way, which result in the
+ *  calculations running sequentially
+ */
+def costlyCalculation1(n: Long): Future[Long] =
+  Future {
+    concurrent.blocking(Thread.sleep(n * 1000))
+    n * 4
+  }
+
+def costlyCalculation2(n: Long): Future[Long] =
+  Future {
+    concurrent.blocking(Thread.sleep(n * 1000))
+    n + 3
+  }
+
+@main def run(): Unit =
```

Comment:
Just curious: what does sbt tell you when you invoke the `run` task from within the `week_6` project?

Reply:
```
sbt:week_6> run
[info] Compiling 7 Scala sources to /Users/ericloots/Trainingen/LBT/EPFL/effective-scala-slide-code/week_6/target/scala-0.27/classes ...
[warn] Multiple main classes detected.  Run 'show discoveredMainClasses' to see the list

Multiple main classes detected, select one to run:

 [1] org.epfl.errorhandling.part1.run
 [2] org.epfl.errorhandling.part8.run
 [3] org.epfl.errorhandling.part9.run
 [4] org.epfl.future.sequence.fail.fast.run
 [5] org.epfl.parallel.run



[info] running org.epfl.future.sequence.fail.fast.run
Starting some computation...
Failed: java.lang.Exception: Bad stuff happens when n == 45
[success] Total time: 22 s, completed 20 Oct 2020, 17:05:04
```